src/components/Table/types.ts

```diff
@@ -0,0 +1,135 @@
+import { ReactElement, ReactNode } from "react";
+import { GridCellContent } from "src/components/Table/components";
+import { GridTableApi } from "src/components/Table/GridTableApi";
+import { Margin, Xss } from "src/Css";
+
+export type Kinded = { kind: string };
```

Comment:
Suggestion: Fwiw I'm not a huge fan of `types.ts`-only-types files; would generally prefer to have the types live closest to the code that primarily owns it, like probably `GridTableProsp` for most of these first few...

Disclaimer would be true utility types, like `DiscriminateUnion`.

Reply:
Sounds good. I'm indifferent on it. I like the push to put more related types with the implementation that uses those types, such as `GridDataRow` moves into `Row.tsx` and maybe `GridColumn` moves into `GridTable`? I wasn't sure where to put everything yet. Though, I followed a pattern of moving components to their own files + the associated interfaces. If there is a method/type used my multiple files, then put it in a "shared" area. I'm happy to keep reworking this, but some of this stuff isn't really straight forward on where it should go. At least this is a start.

Reply:
> but some of this stuff isn't really straight forward on where it should go. At least this is a start.

Yeah, 100% it had grown organically for quite awhile now; appreciate the clean up pass!